pkg/cli/validate.go

```diff
@@ -0,0 +1,191 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/client-lib/go/client"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+// validateCmdOptions holds all the inputs and flags for the validate subcommand.
+type validateCmdOptions struct {
+	config *config.CLIConfig
+
+	token string
+}
+
+// newValidateCmd creates a new subcommand for validating tokens.
+func newValidateCmd(cfg *config.CLIConfig) *cobra.Command {
+	opts := &validateCmdOptions{
+		config: cfg,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Validate the input token",
+		Long: strings.Trim(`
+Validate the given justification token passed via token flag or pipe.
+The output will be a tablized view of validation result, token justification
+and standard claims put together, or any errors that occured.
+
+For example:
+
+    # Validate the justification token string
+    jvsctl validate --token "example token string"
+
+    # Validate the justification token read from pipe
+    cat token.txt | jvsctl token --token -
+
+    # Output
+	-------RESULT-------
+	validated!
+	breakglass  false
+	----JUSTIFICATION----
+	explanation  "test"
+	---STANDARD CLAIMS---
+	aud  ["dev.abcxyz.jvs"]
+	iat  "2022-01-01T00:00:00Z"
+	iss  "jvsctl"
+	jti  "test-jwt"
+	nbf  "2022-01-01T00:00:00Z"
+	sub  "jvsctl"
+`, "\n"),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runValidateCmd(cmd, opts, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.token, "token", "t", "",
+		"The JVS token that needs validation")
+	cmd.MarkFlagRequired("token") //nolint // not expect err
+
+	return cmd
+}
+
+func runValidateCmd(cmd *cobra.Command, opts *validateCmdOptions, args []string) error {
+	ctx := context.Background()
+
+	jvsclient, err := client.NewJVSClient(ctx, &client.JVSConfig{
+		Version:         "1",
+		JWKSEndpoint:    opts.config.GetJWKSEndpoint(),
+		CacheTimeout:    5 * time.Minute,
+		AllowBreakglass: true,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create jvs client: %w", err)
+	}
+
+	if opts.token == "-" {
+		stdin, ok := cmd.InOrStdin().(*os.File)
+		// Default to os.Stdin when input is not of type os.File
+		if !ok {
+			stdin = os.Stdin
+		}
+		stat, _ := stdin.Stat()
+		// Check whether the input is from pipe
+		if (stat.Mode() & os.ModeCharDevice) == 0 {
+			// Read from pipe
+			buf, err := io.ReadAll(io.LimitReader(stdin, 64*1_000))
+			if err != nil {
+				return fmt.Errorf("failed to read from pipe: %w", err)
+			}
+			opts.token = string(buf)
+		} else {
+			// Use user input
+			fmt.Print("Enter token: ")
+			fmt.Scanf("%s", &opts.token)
+		}
+	}
+
+	// Validate token
+	breakglass := false
```

Comment:
Can you break the following logic to be 3 functions? 1. print breakglass 2. print justifications 3. print standard claims

Reply:
Created 2. & 3. helper functions, keeping 1 as it is since it is simple enough.